lib/smart_answer/calculators/maternity_paternity_pay_leave_calculator.rb

```diff
@@ -121,7 +121,11 @@ def mother_earnings_employment?
 
     # Earnings and employment test
     def earnings_employment(earnings_employment, work_employment)
-      earnings_employment == "yes" && work_employment == "yes"
+      if employment_status_of_mother == "self-employed"
+        work_employment == "yes"
+      else
+        earnings_employment == "yes" && work_employment == "yes"
+      end
```

Comment:
I think this could be simplified to:

```ruby
return false unless work_employment == "yes"

employment_status_of_mother == "self-employed" || earnings_employment == "yes"
```